swiftnav/src/signal/consts.rs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2025 Swift Navigation Inc.
+// Contact: Swift Navigation <dev@swiftnav.com>
+//
+// This source is subject to the license found in the file 'LICENSE' which must
+// be be distributed together with this source. All other rights reserved.
+//
+// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+// EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+// WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+use crate::math::compile_time_max_u16;
+
+/// Total number of satellites in the GPS constellation.
+pub const NUM_SATS_GPS: u16 = 32;
+/// Total number of satellites in the SBAS constellation.
+pub const NUM_SATS_SBAS: u16 = 19;
+/// Total number of satellites in the GLONASS constellation.
+/// refer to <https://igscb.jpl.nasa.gov/pipermail/igsmail/2012/007771.html> and
+/// <https://igscb.jpl.nasa.gov/pipermail/igsmail/2015/008391.html>
+pub const NUM_SATS_GLO: u16 = 28;
+/// Total number of satellites in the BeiDou constellation.
+pub const NUM_SATS_BDS: u16 = 64;
+/// Total number of satellites in the Galileo constellation.
+pub const NUM_SATS_GAL: u16 = 36;
+/// Total number of satellites in the QZSS constellation.
+pub const NUM_SATS_QZS: u16 = 10;
+
+/// Total number of satellites in all constellations
+pub const NUM_SATS: u16 =
+    NUM_SATS_GPS + NUM_SATS_SBAS + NUM_SATS_GLO + NUM_SATS_BDS + NUM_SATS_QZS + NUM_SATS_GAL;
+
+/// The maximum number of satellites in a single constellation
+pub const MAX_NUM_SATS: u16 = compile_time_max_u16(
+    NUM_SATS_GPS,
+    compile_time_max_u16(
```

Comment:
Interesting way to avoid lazy execution :smile:

Reply:
It seems like `const` traits are [still being worked on](https://github.com/rust-lang/rust/issues/143874) so there isn't any generic `const` comparisons in the standard library. I did find a couple of crates which provide this via proc macros, but this seemed the most straight forward option.